<launch>
  <arg name="debug"               default="false"/>
  <arg name="bag_filename"        default="$(env HOME)/bags/campusrover_bag/20200409_itc_3f_2s.bag"/>
  <!-- ros input topics -->
  <arg name="only_use_velodyne"        default="false"/>
  <arg if="$(arg only_use_velodyne)"  name="pointcloud" default="/velodyne_points_costmap"/>
  <arg unless="$(arg only_use_velodyne)"  name="pointcloud" default="laser_cloud"/>

  <group if="$(arg debug)">
    <param name="use_sim_time" value="true" />
    <!-- <node if="$(arg viz)" name="rviz" pkg="rviz" type="rviz" required="true"
        args="-d $(find cartographer_ros)/configuration_files/demo_2d.rviz" /> -->

    <node name="playbag" pkg="rosbag" type="play"
          args="--clock $(arg bag_filename)" />

    <param name="robot_description"
           textfile="$(find campusrover_description)/urdf/campusrover.urdf" />
    <node name="robot_state_publisher" pkg="robot_state_publisher" 
          type="robot_state_publisher" />

    <node pkg="tf2_ros" type="static_transform_publisher"
          name="map_broadcaster" args="0 0 0 0 0 0 1 /map /base_link" />
    
    <include file="$(find campusrover_sensors)/launch/ira_laser.launch" />

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find campusrover_costmap)/rviz/campusrover_costmap.rviz" />
  </group>

  <node name="campusrover_costmap" pkg="campusrover_costmap" type="local_costmap_node" output="screen">
    <remap from="points2"             to="$(arg pointcloud)" />
    <param name="pcl_viz"             type="bool"   value="false" />
    <param name="map_size_width"      type="double" value="6" />
    <param name="map_size_height"     type="double" value="6" />
    <param name="map_resolution"      type="double" value="0.02" />
    <param name="max_obstacle_height" type="double" value="2.0" />
    <param name="min_obstacle_height" type="double" value="-1.2" />
    <param name="inflation_radius"    type="double" value="0.02" />
    <param name="cost_scaling_factor" type="double" value="10.0" />
  </node>

  <node unless="$(arg only_use_velodyne)" name="laser_to_pointcloud2" pkg="campusrover_costmap" type="laser_to_pointcloud2">
    <remap from="laser" to="/scan" />
    <remap from="pointcloud" to="laser_cloud" />
  </node>

  <!-- <node unless="$(arg only_use_velodyne)" name="elevator_costmap" pkg="campusrover_costmap" type="elevator_costmap">
    <remap from="laser" to="/scan" />
    <remap from="velodyne" to="/velodyne_points" />
    <remap from="pointcloud" to="laser_cloud" />
  </node> -->

</launch>